relay/relay.go

```diff
@@ -99,14 +100,24 @@ type Relay struct {
 	pollingHandlers      []io.Closer
 }
 
-func NewRelay(config Config, flagsConfig ContextFlagsConfig, name string) (*Relay, error) {
+// ArgsRelayer is the DTO used in the relayer constructor
+type ArgsRelayer struct {
+	Config      Config
+	FlagsConfig ContextFlagsConfig
+	Name        string
+	Proxy       bridge.ElrondProxy
```

Comment:
:+1: for the Proxy
The `EthClient` and `EthInstance` should change to interfaces